src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,140 @@
+     * @throws DukeException if user input is invalid
+     */
+    private void parseCreateTask(String[] splitInput, TaskList tasks) throws DukeException {
+        int len = splitInput.length;
+        if (len == 0) {
+            throw new DukeException("No input!");
+        } else if (splitInput[0].equals("todo")) {
+            tasks.editTaskList("todo", getTaskField(splitInput, 1, len), "");
+        } else if (splitInput[0].equals("deadline")) {
+            int index = getDateIndex(splitInput);
+            String taskDescription = getTaskField(splitInput, 1, index);
+            String date = getTaskField(splitInput, index + 1, len);
+            tasks.editTaskList("deadline", taskDescription, date);
+        } else if (splitInput[0].equals("event")) {
+            int index = getDateIndex(splitInput);
+            String taskDescription = getTaskField(splitInput, 1, index);
+            String date = getTaskField(splitInput, index + 1, len);
+            tasks.editTaskList("event", taskDescription, date);
+        } else {
+            throw new DukeException("Duke: OH NO!!! I'm sorry, but I don't know what that means :-(");
+        }
+    }
+
+    /**
+     * Searches for tasks in the <code>TaskList</code> with the input keyword.
+     *
+     * @param splitInput array of user input words
+     * @param tasks list of stored tasks
+     * @throws DukeException if user input does not have a keyword
+     */
+    private void parseFind(String[] splitInput, TaskList tasks) throws DukeException {
+        int len = splitInput.length;
+        if (len == 1) {
+            throw new DukeException("Duke: To search for tasks, enter a keyword");
+        }
+        String keyword = String.join(" ", Arrays.copyOfRange(splitInput, 1, len));
+        tasks.find(keyword);
+    }
+
+    /**
+     * Interprets and executes the input command from the user.
+     *
+     * @param userInput input from user
+     * @param tasks list of stored tasks
+     * @throws DukeException if command does not follow the format
+     */
+    public void parse(String userInput, TaskList tasks) throws DukeException {
+        String[] splitInput = userInput.split(" ");
+        if (userInput.equals("bye")) {
+            isExit = true;
+        } else if (userInput.equals("list")) {
+            Ui.printTaskList(tasks);
+        } else if(splitInput[0].equals("find")) {
+            parseFind(splitInput, tasks);
+        } else if (splitInput[0].equals("mark")) {
+            parseEdit("mark", splitInput, tasks);
+        } else if (splitInput[0].equals("unmark")) {
+            parseEdit("unmark", splitInput, tasks);
+        } else if (splitInput[0].equals("delete")) {
+            parseEdit("delete", splitInput, tasks);
+        } else if (splitInput[0].equals("delete")) {
+            throw new DukeException("Duke: To delete tasks, indicate the index of the task using an integer!");
+        } else {
+            parseCreateTask(splitInput, tasks);
```

Comment:
I like your method namings. It is in verb form and easy to understand what the parser is doing. Good job! 😄